jni/java/src/main/java/com/github/dart_lang/jni/PortProxy.java

```diff
@@ -62,21 +62,28 @@ private static void appendType(StringBuilder descriptor, Class<?> type) {
 
   public static Object newInstance(String binaryName, long port, long threadId, long functionPtr)
       throws ClassNotFoundException {
-    Class clazz = Class.forName(binaryName);
+    Class<?> clazz = Class.forName(binaryName);
     return Proxy.newProxyInstance(
         clazz.getClassLoader(), new Class[] {clazz}, new PortProxy(port, threadId, functionPtr));
   }
 
   @Override
-  public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
-    return _invoke(port, threadId, functionPtr, proxy, getDescriptor(method), args);
+  public Object invoke(Object proxy, Method method, Object[] args) {
+    Object[] result = _invoke(port, threadId, functionPtr, proxy, getDescriptor(method), args);
+    _cleanUp((Long) result[0]);
```

Comment:
Why does calling two native methods instead of 1 make a difference? Shouldn't we be able to do in a single C function what we can do in two?

Reply:
Explained below.